example/objc/main_darwin.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build example
+
+package main
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/ebitengine/purego/objc"
+)
+
+type barObject struct {
+	isa objc.Class `objc:"BarObject:NSObject <NSDelegateWindow>"`
```

Comment:
Or

```go
ClassName() (className string, superClassName string, protocols []string)
```

Reply:
You'd still need `Selector(string) SEL` as well so that's 4 methods that need to be implemented. Two of which are just returning constant strings presumably. Seems like a lot of boilerplate when one single tag and a Selector method is sufficient. This tag can be copied directly from the ported Objective-C code which makes it even less code to write.

The original idea comes from [macdriver](https://github.com/progrium/macdriver) which uses tags. I feel it's an elegant design that mimics the style of original Objective-C code without being a hack.

Reply:
What do you think about returning three values from one method?

Reply:
That's not a bad idea. What would we call the interface since `Selector` doesn't apply now that it has another method? `ClassDefiner`? `Specifier`? `Implementor`? I don't really like any of those. Also, macdriver uses tags to define IBOutlets but I'm not sure if we ever plan on supporting those using RegisterClass?

Reply:
> What would we call the interface since Selector doesn't apply now that it has another method? ClassDefiner? Specifier? Implementor?

Just `Class`? Hmm...

OK, let's go with a tag and revisit when we come up with a better idea. It's still v0 so we can change this.
>